clang/test/CXX/drs/cwg22xx.cpp

```diff
@@ -196,6 +196,16 @@ void g() {
 #endif
 } // namespace cwg2277
 
+namespace cwg2285 { // cwg2285: 4
```

Comment:
It seems that Clang 5 was the first release which exhibits the correct behavior: https://godbolt.org/z/qabGrdvPq

Reply:
You're right, Clang 4 behaves as expected without `#if __cplusplus`.

> Do I need to add this note in test file?

Yes, leave a comment somewhere that Clang 4 doesn't pass the test because of the wrong value of `__cplusplus`, even though it implements the DR.